### PR TITLE
Billing history tab

### PR DESCRIPTION
**Pull request by hovex023**
What:  To implement the Billing History Portion of the new Billing flow/redesign

Why: To create a better user experience

Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [x] NEW: Are there any Satellite database migrations? Are they forwards _and_ backwards compatible? 
 - [x] Does the PR describe what changes are being made?
 - [x] Does the PR describe why the changes are being made?
 - [x] Does the code follow [our style guide](https://github.com/storj/docs/blob/main/code/Style.md)?
 - [x] Does the code follow [our testing guide](https://github.com/storj/docs/blob/main/code/Testing.md)?
 - [x] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [x] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [x] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [x] Does any documentation need updating?
 - [x] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+     */
+    public get getSortDirection(): SortDirection {
+        return this.sortDirection === SortDirection.DESCENDING ? SortDirection.ASCENDING : SortDirection.DESCENDING;
+    }
+
+    public get areAccessGrantsSortedByName(): boolean {
+        return this.sortBy === AccessGrantsOrderBy.NAME;
+    }
+
+    /**
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue:55]: wrong jsdoc position?

File: web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import BillingHistoryShape from '@/components/account/billing/billingTabs/BillingHistoryShape.vue';
+import { PaymentsHistoryItem, PaymentsHistoryItemType } from '@/types/payments';
+
+// @vue/component
+@Component({
+    components: {
+        VList,
+        BillingHistoryHeader,
+        BillingHistoryShape,
+        VPagination
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:30]: trailing comma

File: web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    public get billingHistoryStructure(): any {
+        return BillingHistoryShape;
+    }
+}
+</script>
+
+<style scoped lang="scss">
+     .billing_history2 {
+            position: relative;
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:48]: 7 spaces, should be 4

File: web/satellite/src/components/account/billing/billingTabs/Coupons.vue [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+@Component({
+    components: {
+    },
+})
+export default class BillingArea extends Vue {
+
+}
+</script>
+
+<style scoped lang="scss">
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/account/billing/billingTabs/Coupons.vue:25]: no empty tags

**Review by NikolaiYurchenko — CHANGES_REQUESTED** [review]
and resolve merge conflict

File: web/satellite/src/components/account/billing/billingTabs/Coupons.vue [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        <router-view />
+    </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue } from 'vue-property-decorator';
+import { RouteConfig } from '@/router';
+
+// @vue/component
+@Component({
```
[the remark this answers is not in the document]
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/Coupons.vue:15]: image
Looks like linter wants it to be formatted that way

**Review by NikolaiYurchenko — DISMISSED** [review]
generally LG

File: web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue [hunk 6]
```diff
@@ -1,22 +1,52 @@
...
 @Component({
     components: {
+        VList,
+        BillingHistoryHeader,
     },
 })
+
 export default class BillingArea extends Vue {
+    public get historyItems(): PaymentsHistoryItem[] {
+        return this.$store.state.paymentsModule.paymentsHistorly;
```
Comment (mobyvb) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue:32]: typo "paymentsHistorly"

**Review by mobyvb — COMMENTED** [review]
looks good overall but I noticed a typo - I can approve when this is fixed
Another thing I noticed - you uploaded both svgs and pngs, but in the code, only the svgs are used. Are the pngs necessary?
I also saw you flipped one use of `isNewAccessGrantFlow` - was this intentional?

File: web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue (hunk 6 above)
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/BillingHistory.vue]: fixed

**Comment by hovex023** [comment]
> looks good overall but I noticed a typo - I can approve when this is fixed Another thing I noticed - you uploaded both svgs and pngs, but in the code, only the svgs are used. Are the pngs necessary? I also saw you flipped one use of `isNewAccessGrantFlow` - was this intentional?

Removed the typo.  Also removed the unnecessary png files. The linter was complaining about those flags for some reason. Looking at the access grant section, I believe the flags are correct right now

**Review by NikolaiYurchenko — DISMISSED** [review]

File: web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue [hunk 7]
```diff
@@ -0,0 +1,145 @@
+// Copyright (C) 2022 Storj Labs, Inc.
```
Comment (mobyvb) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistorHeader.vue:1]: nit - typo in filename

File: web/satellite/src/components/account/billing/billingTabs/BillingHistoryShape.vue [hunk 8]
```diff
@@ -0,0 +1,228 @@
...
+import Calendar from '@/../static/images/billing/calendar.svg';
+
+// @vue/component
+@Component({
+    components: {
+        Calendar,
+        CheckIcon,
+    },
+})
+export default class BillingHistoryShape extends Vue {
```
Comment (mobyvb) [inline web/satellite/src/components/account/billing/billingTabs/BillingHistoryShape.vue:49]: Why is this component called BillingHistoryShape?

File: web/satellite/src/components/accessGrants/AccessGrants.vue [hunk 9]
```diff
@@ -169,7 +169,7 @@
                 </span>
             </div>
         </div>
-        <div v-if="!isNewAccessGrantFlow">
```
Comment (mobyvb) [inline web/satellite/src/components/accessGrants/AccessGrants.vue:172]: why is this flag flipped? With this change, the old access flow no longer properly lists accesses. This is the only comment blocking my approval. Everything else looks good to me, and I'm happy to approve once the old access grants flow works on this branch

Comment (hovex023) [reply web/satellite/src/components/accessGrants/AccessGrants.vue:172]: fixed

File: web/satellite/src/components/account/billing/billingTabs/BillingHistoryShape.vue (hunk 8 above)
Comment (hovex023) [reply web/satellite/src/components/account/billing/billingTabs/BillingHistoryShape.vue:49]: Vlist takes a dom shape as a property. So I named it BillinghistoryShape as it's passed into Vlist.

**Review by NikolaiYurchenko — APPROVED** [review]

**Review by mobyvb — APPROVED** [review]

**Pull request merged**